packages/fx-core/src/plugins/resource/apiconnector/plugin.ts

```diff
@@ -41,52 +36,55 @@ export class ApiConnectorImpl {
     const projectPath = inputs.projectPath;
     const languageType: string = ctx.projectSetting!.programmingLanguage!;
     const config: ApiConnectorConfiguration = this.getUserDataFromInputs(inputs);
-    for (const componentItem of config.ComponentPath) {
-      await this.scaffoldInComponent(projectPath, componentItem, config, languageType);
-    }
-    return ResultFactory.Success();
-  }
-
-  private async scaffoldInComponent(
-    projectPath: string,
-    componentItem: string,
-    config: ApiConnectorConfiguration,
-    languageType: string
-  ): Promise<ApiConnectorResult> {
-    const componentPath = path.join(projectPath, componentItem);
+    // backup relative files.
     const backupFolderName = generateTempFolder();
-    const sampleFileName = getSampleCodeFileName(config.APIName, languageType);
-    await this.backupExistingFiles(componentPath, sampleFileName, backupFolderName);
+    const componentFolders: string[] = [];
+    config.ComponentPath.forEach((item) => componentFolders.push(path.join(projectPath, item)));
+    componentFolders.forEach(async (componentPath) => {
+      await this.backupExistingFiles(componentPath, backupFolderName);
```

Comment:
some functions take `componentPath` as input and some functions take `projectPath` and `componentItem` as input. Since `componentPath`=`projectPath`+`componentItem`, can we make some small refractors to align the input parameters?